lib/src/isolate_manager/isolate_manager.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:async';
+
+import 'package:dash_kit_network/dash_kit_network.dart';
+
+// This is an abstract class for managing isolates It defines the basic methods
+// needed to start, stop and send messages to an isolate. Flutter Web doesn't
+// support isolates so the `IsolateManager` class has a different implementation
+// for web.
+// ignore: prefer-match-file-name
+abstract class IsolateManagerInterface {
+  // Method for starting the isolate. Should be called before sending messages.
+  Future<void> start();
+
+  // Method for stopping the isolate. Should be called when the isolate is no
+  //longer needed.
+  void stop();
+
+  // Method for sending messages to the isolate. The `mapper` function is
+  // executed in the isolate and the result is returned.
+  Future<FutureOr<T>> send<T>(
+    Response<dynamic> response,
+    FutureOr<T> Function(Response<dynamic>) mapper,
+  );
+}
+
+// Class Task is used to store the data needed to send a message to the isolate.
+class Task<T> {
+  const Task(
+    this.id,
+    this.response,
+    this.mapper,
+  );
+
+  final String id;
+  final Response<dynamic> response;
+  final T Function(Response<dynamic>) mapper;
+
+  @override
+  String toString() {
+    return 'Task{id: $id, response: $response, mapper: $mapper}';
+  }
+}
+
+// Class Result is used to store the result of the mapper function executed in
+// the isolate.
+class Result<T> {
+  const Result({
+    required this.id,
+    this.result,
+    this.error,
+  });
+
+  final String id;
+  final T? result;
+  final dynamic error;
+
+  @override
+  String toString() {
+    return 'Result{id: $id, result: $result, error: $error}';
+  }
+}
+
+// The IsolateStatus enum is used to track the status of the isolate.
+enum IsolateStatus {
+  created,
+  initializing,
```

Comment:
do we need to add the error state during initializing? What do you think?

Reply:
There is no need for this status in the current implementation